File: src/tests/fidl/source_compatibility/protocol-event-add/go/step_03_after.go
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package main

import (
	"context"
	lib "fidl/fidl/test/protocoleventadd"
)

// [START contents]
func expectEvents(c *lib.ExampleWithCtxInterface) {
	_ = c.ExpectOnExistingEvent(context.Background())
	_ = c.ExpectOnNewEvent(context.Background())
}

func sendEvents(p *lib.ExampleEventProxy) {
	_ = p.OnExistingEvent()
	_ = p.OnNewEvent()
}

// [END contents]

func main() {}

File: src/tests/fidl/source_compatibility/protocol-event-add/dart/step_03_after.dart
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// false positive, see https://github.com/dart-lang/linter/issues/1381
// ignore_for_file: close_sinks
import 'dart:async';
import 'package:fidl_fidl_test_protocoleventadd/fidl_async.dart' as fidllib;

// [START contents]
class Server extends fidllib.Example {
  final _onExistingEventStreamController = StreamController<void>();
  final _onNewEventStreamController = StreamController<void>();

  @override
  Stream<void> get onExistingEvent => _onExistingEventStreamController.stream;

  @override
  Stream<void> get onNewEvent => _onNewEventStreamController.stream;
}

void expectEvents(fidllib.ExampleProxy client) async {
  await client.onExistingEvent.first;
  await client.onNewEvent.first;
}
// [END contents]
